x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph/graph.tsx

```diff
@@ -82,97 +83,106 @@ const edgeTypes = {
-        edgeTypes={edgeTypes}
-        nodes={nodesState}
-        edges={edgesState}
-        onNodesChange={onNodesChange}
-        onEdgesChange={onEdgesChange}
-        proOptions={{ hideAttribution: true }}
-        panOnDrag={isGraphInteractive && !isLocked}
-        zoomOnScroll={isGraphInteractive && !isLocked}
-        zoomOnPinch={isGraphInteractive && !isLocked}
-        zoomOnDoubleClick={isGraphInteractive && !isLocked}
-        preventScrolling={interactive}
-        nodesDraggable={interactive && isGraphInteractive && !isLocked}
-        maxZoom={1.3}
-        minZoom={0.1}
-      >
-        {interactive && <Controls onInteractiveChange={onInteractiveStateChange} />}
-        <Background id={backgroundId} />
-      </ReactFlow>
-    </div>
-  );
-};
+    }, [nodes, edges, setNodes, setEdges, isGraphInteractive]);
+
+    const onInteractiveStateChange = useCallback(
+      (interactiveStatus: boolean): void => {
+        setIsGraphInteractive(interactiveStatus);
+        setNodes((currNodes) =>
+          currNodes.map((node) => ({
+            ...node,
+            data: {
+              ...node.data,
+              interactive: interactiveStatus,
+            },
+          }))
+        );
+      },
+      [setNodes]
+    );
+
+    const onInitCallback = useCallback(
+      (xyflow: ReactFlowInstance<Node<NodeViewModel>, Edge<EdgeViewModel>>) => {
+        window.requestAnimationFrame(() => xyflow.fitView());
+        fitViewRef.current = xyflow.fitView;
+
+        // When the graph is not initialized as interactive, we need to fit the view on resize
+        if (!interactive) {
+          const resizeObserver = new ResizeObserver(() => {
+            xyflow.fitView();
+          });
+          resizeObserver.observe(document.querySelector('.react-flow') as Element);
+          return () => resizeObserver.disconnect();
+        }
+      },
+      [interactive]
+    );
+
+    return (
+      <div {...rest}>
+        <SvgDefsMarker />
+        <ReactFlow
+          fitView={true}
+          onInit={onInitCallback}
+          nodeTypes={nodeTypes}
+          edgeTypes={edgeTypes}
+          nodes={nodesState}
+          edges={edgesState}
+          nodesConnectable={false}
+          edgesFocusable={false}
+          onlyRenderVisibleElements={ONLY_RENDER_VISIBLE_ELEMENTS}
+          snapToGrid={true} // Snap to grid is enabled to avoid sub-pixel positioning
+          snapGrid={[1, 1]}
+          onNodesChange={onNodesChange}
+          onEdgesChange={onEdgesChange}
+          proOptions={{ hideAttribution: true }}
+          panOnDrag={isGraphInteractive && !isLocked}
+          zoomOnScroll={isGraphInteractive && !isLocked}
+          zoomOnPinch={isGraphInteractive && !isLocked}
+          zoomOnDoubleClick={isGraphInteractive && !isLocked}
+          preventScrolling={interactive}
+          nodesDraggable={interactive && isGraphInteractive && !isLocked}
+          maxZoom={1.3}
+          minZoom={0.1}
```

Comment:
not important right now but in general there are a lot of settings scattered around which might be better to export as consts from one file or object, to make it easier to configure in the future if needed